### ISSUE TITLE
Add doctests for methods in SpacingHelper

### ISSUE THREAD

**Issue by adtac** [coala-analyzer/coala#2765]
Very suitable IMO

**Comment by jayvdb** [comment]
Hold that thought!

IMO `SpacingHelper` should be destroyed with fire. #2769

**Comment by sils1297** [comment]
I agree. IMO the class is badly designed (it's a bunch of functions trying to be an object) and when I wrote it I had no idea of the existence of expandtabs. If we can destroy it with fire that's awesome.

FWIW the other related issues are:

- https://github.com/coala/coala/issues/2769
- https://github.com/coala/coala/issues/2756

**Comment by jayvdb** [comment]
Can you give me a pointer to how you are deprecating things in coala?
I think I saw you say somewhere that your deprecation system isnt fully developed.
If so, I could write up a design doc, explaining the Pywikibot system I have been involved in building.
https://github.com/wikimedia/pywikibot-core/blob/master/pywikibot/tools/__init__.py#L1100

**Comment by sils1297** [comment]
I'm on phone, for design docs we have the ceps repo. We only have the settings deprecation decorator so far I think.

**Comment by Makman2** [comment]
@jayvdb maybe we just keep the old `SpacingHelper` class while you do your rewrite/improvements in another file. And inside `SpaingHelper.__init__` we print a deprecation warning as well in every `staticmethod` imo :)

**Comment by sils1297** [comment]
Lets at least make a simple decorator otherwise you'll start forgetting
things and duplicate the warning everywhere

**Comment by Makman2** [comment]
@sils1297 right :+1: :)

**Comment by jayvdb** [comment]
I see we have [`coalib/bearlib/__init__.py`](https://github.com/coala/coala/blob/master/coalib/bearlib/__init__.py) `@deprecate_settings` , which is (not surprisingly) very similar to one I have developed with previously.  All good.

**Comment by sils1297** [comment]
Yep we just need one for bears as well that warns upon class instanciation or so

**Comment by sils1297** [comment]
and one for functions outside bears so we can deprecate coala API
